source/client.c

```diff
@@ -2704,7 +2761,8 @@ static enum aws_mqtt_client_request_state s_publish_send(uint16_t packet_id, boo
         if (!timeout_task_arg) {
             return AWS_MQTT_CLIENT_REQUEST_ERROR;
         }
-        task_arg->timeout_arg = timeout_task_arg;
+        task_arg->timeout_wrapper.timeout_task_arg = timeout_task_arg;
```

Comment:
does this make any sense? add a comment like this, but making sense:

/* The request and its timeout task point to each other. Whichever one finishes first will null out both pointers. Then when the second one runs, it will know that it wasn't first */